ipns/validate_test.go

```diff
@@ -413,3 +412,38 @@ func genKeys(t *testing.T) (crypto.PrivKey, peer.ID, string) {
 
 	return priv, pid, ipnsKey
 }
+
+func TestValidateWithPeerID(t *testing.T) {
+	path := []byte("/ipfs/bafkreifjjcie6lypi6ny7amxnfftagclbuxndqonfipmb64f2km2devei4")
+	eol := time.Now().Add(time.Hour)
+
+	rnd := rand.New(rand.NewSource(42))
+
+	sk, pk, err := crypto.GenerateEd25519Key(rnd)
+	assert.Nil(t, err)
+
+	pid, err := peer.IDFromPublicKey(pk)
+	assert.Nil(t, err)
+
+	entry, err := Create(sk, path, 1, eol, 0)
+	assert.Nil(t, err)
+
+	t.Run("valid peer ID", func(t *testing.T) {
+		t.Parallel()
+		err = ValidateWithPeerID(pid, entry)
+		assert.Nil(t, err)
+	})
+
+	t.Run("invalid peer ID", func(t *testing.T) {
+		t.Parallel()
+
+		_, pk2, err := crypto.GenerateEd25519Key(rnd)
+		assert.Nil(t, err)
+
+		pid2, err := peer.IDFromPublicKey(pk2)
+		assert.Nil(t, err)
+
+		err = ValidateWithPeerID(pid2, entry)
+		assert.NotNil(t, err)
```

Comment:
Please check the actual error returned (I guess this error type or var is public and we can `errors.Is` it).